Script/update_grovepi.sh

```diff
@@ -5,34 +5,147 @@ PIHOME=/home/pi
+echo " |_____|_| |_|\__,_|\__,_|___/\__|_|  |_|\___||___/ ";
+echo "                                                    ";
+echo "                                                    ";
+echo "  _____                    _____ _ "
+echo " / ____|                  |  __ (_)  "
+echo "| |  __ _ __ _____   _____| |__) |   "
+echo "| | |_ | '__/ _ \ \ / / _ \  ___/ |  "
+echo "| |__| | | | (_) \ V /  __/ |   | |  "
+echo " \_____|_|  \___/ \_/ \___|_|   |_|  "
+echo " "
+
+# show some feedback on the console
+if [ -f $DEXTERSCRIPT/functions_library.sh ]; then
+  source $DEXTERSCRIPT/functions_library.sh
+  feedback "Welcome to GrovePi Installer."
 else
-	echo "Cloning"
-    cd $PIHOME/$DEXTER/
-    git clone https://github.com/DexterInd/GrovePi
-    cd GrovePi
+  echo "Welcome to GrovePi Installer."
+fi
+echo "Updating GrovePi for $selectedbranch branch with the following options:"
+([[ $installdependencies = "true" ]] && echo "  --no-dependencies=false") || echo "  --no-dependencies=true"
+([[ $updaterepo = "true" ]] && echo "  --no-update-aptget=false") || echo "  --no-update-aptget=true"
+([[ $install_pkg_scriptools = "true" ]] && echo "  --bypass-pkg-scriptools=false") || echo "  --bypass-pkg-scriptools=true"
+echo "  --user-local=$userlocal"
+echo "  --env-local=$envlocal"
+echo "  --system-wide=$systemwide"
+
+# in case the following packages are not installed and `--no-dependencies` option has been used
+if [[ $installdependencies = "false" ]]; then
+  command -v git >/dev/null 2>&1 || { echo "I require git but it's not installed. Don't use --no-dependencies option. Aborting." >&2; exit 1; }
+  command -v python >/dev/null 2>&1 || { echo "Executable \"python\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 2; }
+  command -v python3 >/dev/null 2>&1 || { echo "Executable \"python3\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 3; }
+  command -v pip >/dev/null 2>&1 || { echo "Executable \"pip\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 4; }
+  command -v pip3 >/dev/null 2>&1 || { echo "Executable \"pip3\" couldn't be found. Don't use --no-dependencies option. Aborting." >&2; exit 5; }
 fi
 
-change_branch $BRANCH   # Change to a branch we're working on in the GrovePi Directory. 
-                        # Variable $BRANCH comes from /upd_script/fetch.sh
+# create rest of list of arguments for script_tools call
+optionslist+=("$selectedbranch")
+[[ $usepython3exec = "true" ]] && optionslist+=("--use-python3-exe-too")
+[[ $updaterepo = "true" ]] && optionslist+=("--update-aptget")
+[[ $installdependencies = "true" ]] && optionslist+=("--install-deb-deps")
+[[ $install_pkg_scriptools = "true" ]] && optionslist+=("--install-python-package")
+
+echo "Options used for script_tools script: \"${optionslist[@]}\""
+
+# update script_tools first
+curl --silent -kL dexterindustries.com/update_tools > $PIHOME/tmp_script_tools.sh
```

Comment:
why in a temporary spot??  Just curl it once and for all.

Reply:
If you read the source code, you'll see that's not possible, because `${optionslist[@]}` has to be evaluated first and then they can be used. Since it gets evaluated you end up with something that doesn't look like a command.

Reply:
The curl itself doesn't make use of optionlist. And only the curl is needed to be able to source functions_library.
So you can do it at the top. 
curl
source functions_library
then later on, run the install script.

And I still don't get why it's needed in a tmp folder.

Reply:
I could move the curl command up top and save the temp file. But do we want to do that? It's like the temp file is exist for a bit more time.

On the other hand, the curl command *no longer makes use of optionlist* because we're using the temp file. Now, when passing `${optionslist[@]}` to the curl command through piping, `{optionslist[@]}` gets evaluated and returns exits due to not being an actual command, so that's the reason I used a temp file. 
There may be a way to bypass the evaluation of the variable, but that could take more time to research and instead, I would want to have this out ASAP.